compiler/rustc_middle/src/middle/privacy.rs

```diff
@@ -27,26 +27,148 @@ pub enum AccessLevel {
     Public,
 }
 
+#[derive(Clone, Copy, PartialEq, Eq, Debug, HashStable, Default)]
+pub struct EffectiveVisibility {
+    public: Option<Visibility>,
+    exported: Option<Visibility>,
+    reachable: Option<Visibility>,
+    reachable_from_impl_trait: Option<Visibility>,
+}
+
+impl EffectiveVisibility {
+    fn get_field(&self, tag: AccessLevel) -> &Option<Visibility> {
+        match tag {
+            AccessLevel::Public => &self.public,
+            AccessLevel::Exported => &self.exported,
+            AccessLevel::Reachable => &self.reachable,
+            AccessLevel::ReachableFromImplTrait => &self.reachable_from_impl_trait,
+        }
+    }
+
+    fn get_mut_field(&mut self, tag: AccessLevel) -> &mut Option<Visibility> {
```

Comment:
```suggestion
    fn get_field_mut(&mut self, tag: AccessLevel) -> &mut Option<Visibility> {
```
https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter